### PR TITLE
[11.0][MIG] hr_holidays_imposed_days

### PR DESCRIPTION
**Pull request by astirpe**
Porting of hr_holidays_imposed_days to V11

### CODE REVIEW

File: hr_holidays_imposed_days/models/hr_employee.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        """ add imposed days if a configuration exists """
+
+        employee = super(HrEmployee, self).create(values)
+
+        imposed = self.env['hr.holidays.imposed'].search(
+            employee._get_search_imposed_parameters()
+        )
+        for imposed_day in imposed:
+            created_leave = self.env['hr.holidays'].create({
+                'number_of_days_temp': 1.,
```
Comment (feketemihai) [inline hr_holidays_imposed_days/models/hr_employee.py]: @astirpe Here i think you can grab the number_of_days from imposed days, if you make a imposed holiday for more than one day...and probably to test the duration to e the same..

**Review by feketemihai — CHANGES_REQUESTED** [review]

File: hr_holidays_imposed_days/models/hr_employee.py (hunk 1 above)
Comment (astirpe) [reply hr_holidays_imposed_days/models/hr_employee.py]: Right, thank you!

**Review by feketemihai — APPROVED** [review]

**Comment by feketemihai** [comment]
Code review, not tested...

File: hr_holidays_imposed_days/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Copyright 2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Imposed holidays days',
+    'version': '11.0.1.0.0',
+    'category': 'Human Resources',
+    'license': 'AGPL-3',
+    'author': 'Camptocamp SA, '
+              'Odoo Community Association (OCA)',
+    'website': 'http://www.camptocamp.com',
```
Comment (simahawk) [inline hr_holidays_imposed_days/__manifest__.py]: https://github.com/OCA/hr

File: hr_holidays_imposed_days/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Author: Damien Crier
+# Copyright 2016 Camptocamp SA
```
Comment (simahawk) [inline hr_holidays_imposed_days/__manifest__.py]: 2018

File: hr_holidays_imposed_days/models/hr_holidays_imposed.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
+# Author: Damien Crier
+# Copyright 2016 Camptocamp SA
```
Comment (simahawk) [inline hr_holidays_imposed_days/models/hr_holidays_imposed.py]: 2018

File: hr_holidays_imposed_days/models/hr_holidays_imposed.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                tools.DEFAULT_SERVER_DATETIME_FORMAT) + datetime.timedelta(
+                hours=HOURS_PER_DAY)
+            self.date_to = str(date_to_with_delta)
+
+        # Compute and update the number of days
+        self._set_duration()
+
+    @classmethod
+    def compute_nb_days(self, diff):
+        return round(math.floor(diff))+1
```
Comment (simahawk) [inline hr_holidays_imposed_days/models/hr_holidays_imposed.py]: spaces around "+"

File: hr_holidays_imposed_days/tests/test_hr_holidays_imposed.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
+# Author: Damien Crier
+# Copyright 2016 Camptocamp SA
```
Comment (simahawk) [inline hr_holidays_imposed_days/tests/test_hr_holidays_imposed.py]: 2018

File: hr_holidays_imposed_days/tests/test_hr_holidays_imposed.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+    def setUp(self):
+        super(TestHolidaysImposedDays, self).setUp()
+        # Base models
+        self.employee_model = self.env['hr.employee']
+        self.holiday_status_model = self.env['hr.holidays.status']
+        self.holiday_imposed_model = self.env['hr.holidays.imposed']
+        self.holiday_model = self.env['hr.holidays']
+
+        # create leave types that we will be manupilating
```
Comment (simahawk) [inline hr_holidays_imposed_days/tests/test_hr_holidays_imposed.py]: *manipulating

**Review by simahawk — CHANGES_REQUESTED** [review]
LGTM, some small changes

**Comment by simahawk** [comment]
forgot: squash translations commits ;)

**Comment by simahawk** [comment]
you can squash commits by @leemannd 
image
they are part of the same whole work like "[imp] hr_holidays_imposed_days: misc fix/imp"

**Comment by astirpe** [comment]
Tests fail now, so I will look at them when I can have some time to dedicate :)

**Comment by astirpe** [comment]
Green!

**Review by simahawk — APPROVED** [review]
@astirpe one last thing: if you keep the commit "Tests refactoring" in the merge, please prefix it w/ the module name, like "[imp] hr_holidays_imposed_days: refactor tests" or similar, tnx :wink:

**Comment by astirpe** [comment]
I will squash that commit with the one of the porting to V11, just tell me when this PR is ready to be merged and I will do it immediately ;)

**Review by yung-wang — APPROVED** [review]
We already using this module in our V11 environment. Approved

**Comment by feketemihai** [comment]
@astirpe @yung-wang How is working the module with users timezones and a resource calendar that have more than 8 hours per day, by default if you put the break hour, it gives 9 hours on a daily schedule, so  if you leave the imposed days with 8 hours, it will give you only 0.875 if you change something in the leave, not 1 day...we will see after this moduel and hr_holidays_public (which adds fractional calculation - not quarterly as Odoo is doing now)  are merged...

**Comment by feketemihai** [comment]
@pedrobaeza Do you know why CI raised error on Transifex user password?

**Comment by pedrobaeza** [comment]
Well, Transifex secure was not the same in 10.0 than in 11.0. I have switched to the 10.0 one.

**Pull request merged**